### PR TITLE
Render error page for page errors

### PR DESCRIPTION
**Pull request by rebeccamcginn**
Resolves [WSTEAMA-104](https://jira.dev.bbc.co.uk/browse/WSTEAMA-104)

**Overall change:**
If renderDocument throws an error, call again with error set to true and status set to 500. This returns the error page to the user.

Currently, any error thrown results in the error message being shown to the user, as per this line - https://github.com/bbc/simorgh/blob/latest/src/server/index.jsx#L227

**Code changes:**

- Wrap renderDocument in a try catch block. Log error and call renderDocument again, this time with status set to 500 and error set to true. Our withError page handler then returns our ErrorPage component.
- Remove unnecessary comment

Long term, we'd like to:

- Refactor `getRouteProps` so that it can be moved outside the try block, thus allowing us to access `service` within the existing catch block. This in turn would allow us to remove the nested try catch statement added in this PR and move the second call to `renderDocument` to the existing catch block.
- Create a `renderErrorPage` function that can be called from the existing catch in place of the second call to `renderDocument`. This would require refactoring server/Document.

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by RetiredBlob — APPROVED** [review]

**Review by nicola-d-bbc — APPROVED** [review]

File: src/server/index.jsx [hunk 1]
```diff
@@ -223,7 +251,6 @@ server.get(
         headers,
       });
 
-      // Return an internal server error for any uncaught errors
       res.status(500).send(message);
```
Comment (RetiredBlob) [inline src/server/index.jsx:254]: We should consider only returning the error message if we're running locally - descriptive error messages being shown to the public can be embarrassing and a potential security issue

Comment (andrewscfc) [reply src/server/index.jsx:254]: Yeah I'd definitely hide the error message in production for security reasons primarly.

Comment (rebeccamcginn) [reply src/server/index.jsx:254]: We can add that to the refactor ticket 👍

**Comment by LilyL0u** [comment]
Looks good.

What before looked like this:
Screenshot 2022-06-29 at 16 05 38


Is now this:

Screenshot 2022-06-29 at 16 15 41

**Pull request merged**